### PR TITLE
Provide support for masking HTML tags while editing with real example

### PR DESCRIPTION
**Pull request by tduong992**
From [d.o](https://www.drupal.org/node/2757813), we've started writing a patch to provide this feature.
Now here we will implement a real case with the actual HTML tags. (This PR reverts the [previous one](https://github.com/sasanikolic90/tmgmt_ckeditor/pull/1)).

As first step, I've just added an initial alter hooks to mask TMGMT HTML tags in tmgmt_ckeditor.module. Then started to replace the hardcoded b- and img-tags with some regex and already copied a test. Still work in progress.

This test will not pass yet until two patches are committed ([Add methods in JobItem to get segmented data](https://www.drupal.org/node/2737131) and [Provide support for masking HTML tags while editing](https://www.drupal.org/node/2757813)).

### CODE REVIEW

File: tmgmt_ckeditor.module [hunk 1]
```diff
@@ -57,25 +57,18 @@ function tmgmt_ckeditor_form_tmgmt_job_item_edit_form_alter(&$form, FormStateInt
  * Implements hook_tmgmt_data_item_text_output_alter().
  */
 function tmgmt_ckeditor_tmgmt_data_item_text_output_alter(&$text, $data_item, JobItemInterface $job_item) {
-  if (preg_match_all('/(<([^\btmgmt\b]+[^\s]+)[^>]+?>)([^>]+?[^\/][^>]+>)/', $text, $matches)) {
+  if (preg_match_all('/<((/?[a-z-]+).*?)\s?/?>/', $text, $matches)) {
```
Comment (tduong992) [inline tmgmt_ckeditor.module]: this gives me an error

```
Warning: preg_match_all(): Unknown modifier '?' in tmgmt_ckeditor_tmgmt_data_item_text_output_alter() (line 60 of modules/tmgmt_ckeditor/tmgmt_ckeditor.module).
```

Needs work.

Comment (tduong992) [inline tmgmt_ckeditor.module]: my bad, i've forgotten to escape "/" in the regex. now is fixed in the commit e7f021e

File: src/Tests/TMGMTCKEditorHtmlTagTest.php [hunk 2]
```diff
@@ -60,6 +60,24 @@ public function testReviewForm() {
     $node = $this->drupalGetNodeByTitle('Segments test');
     $this->drupalGet('node/' . $node->id());
 
+    // Node text values.
+    $title = 'Segments test';
+
+    $masked_body = '<tmgmt-segment id="1">This is the first segment.</tmgmt-segment><tmgmt-tag element="br /" raw="&lt;br /&gt;" />';
```
Comment (sasanikolic90) [inline src/Tests/TMGMTCKEditorHtmlTagTest.php]: Should we mask only the tags that are inside the `<tmgmt-segment>` tags or also `<br />` and others that are outside?

**Comment by Berdir** [comment]
@sasanikolic90 Too early, this PR was not updated for the changed hook structure.

**Comment by sasanikolic90** [comment]
@Berdir I updated the hooks in the last commit.

**Pull request merged**